mslib/utils/verify_waypoint_data.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+"""
+
+    mslib.utils.verify_waypoint_data
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    basic checks for xml waypoint data.
+
+    This file is part of MSS.
+
+    :copyright: Copyright 2024 Reimar Bauer
+    :license: APACHE-2.0, see LICENSE for details.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+
+import xml.dom.minidom
+import xml.parsers.expat
+
+
+def verify_waypoint_data(xml_content):
```

Comment:
This could be an ideal use-case for [XML Schemas](https://en.wikipedia.org/wiki/XML_Schema_(W3C)). That would require more than the stdlib xml support.

Reply:
this is currently only a simple solution. A refactoring needs to be done too.